### PR TITLE
Adapter Parameters Written as MetaValues to SearchParameters

### LINKED ISSUE #4878: NovorAdapter tests fail
Due to CI being down, this was not caught but the change for the spectrum index metavalue from float to int makes
the test files not fit anymore.
Be aware, they are only tested in the nightlies since we only check out licensed third-party there.

### PR DESCRIPTION
**Pull request by Waschi97**
- Added a static function to `DefaultParamHandler` that writes a `Param` object to a `MetaInfoInterface`

This allows to replicate a identification search with the same parameters. This will be used for a (hopefully) better DBSuitability calculation later.

- Changed all identification search adapters to use this funtion and write their parameters to the `SearchParameters` of the first `ProteinIdentification`

- Changed the test files of the adapters accordingly

- Also changed Novor test files - resolves #4878

- Added a test for the new function to the `DefaultParamHandler_test`

### CODE REVIEW

File: src/openms/include/OpenMS/DATASTRUCTURES/DefaultParamHandler.h [hunk 1]
```diff
@@ -132,6 +133,17 @@ namespace OpenMS
     /// Non-mutable access to the registered subsections
     const std::vector<String>& getSubsections() const;
 
+    /**
+    * @brief Writes all parameters to meta values
+    *
+    * Parameters are written with 'name' as key and 'value' as value
+    *
+    * @param write_this  Params to be written
+    * @param write_here  a MetaInfoInterface object were the meta values will be written
```
Comment (jpfeuffer) [inline src/openms/include/OpenMS/DATASTRUCTURES/DefaultParamHandler.h:142]:
```suggestion
    * @param write_here  a MetaInfoInterface object into which the meta values will be written 
```

File: src/openms/include/OpenMS/DATASTRUCTURES/DefaultParamHandler.h [hunk 2]
```diff
@@ -132,6 +133,17 @@ namespace OpenMS
...
+    * @param prefix      will be added infront of the parameter name with an ':' as the meta value key
```
Comment (jpfeuffer) [inline src/openms/include/OpenMS/DATASTRUCTURES/DefaultParamHandler.h:143]:
```suggestion
    * @param prefix      will be added in front of the parameter name with a ':' as the meta value key
```

File: src/tests/class_tests/openms/source/DefaultParamHandler_test.cpp [hunk 3]
```diff
@@ -50,148 +51,169 @@ using namespace OpenMS;
...
+START_SECTION(static void writeParametersToMetaValues(const Param& write_this, MetaInfoInterface& write_here, const String& prefix = ""))
+  MetaInfoInterface meta_values;
+  Param p;
+  p.setValue("int", 1);
+  p.setValue("string", "test");
+  p.setValue("ignore:bli", 4711);
+  DefaultParamHandler::writeParametersToMetaValues(p, meta_values);
+  ABORT_IF(!meta_values.metaValueExists("int"))
+  ABORT_IF(!meta_values.metaValueExists("string"))
+  //ABORT_IF(!meta_values.metaValueExists("ignore:bli"))
```
Comment (jpfeuffer) [inline src/tests/class_tests/openms/source/DefaultParamHandler_test.cpp:207]: why commented out?

**Comment by Waschi97** [comment]
merge?

**Comment by cbielow** [comment]
Please wait
Comments coming soon

**Comment by jpfeuffer** [comment]
Feel free to merge.

**Pull request merged**